app/Http/Controllers/Pesan/PesanController.php

```diff
@@ -0,0 +1,271 @@
+        $data->put('page_title', 'Buat Pesan');
+        $data->put('page_description', 'Managemen Pesan');
+        $list_desa = DataDesa::get();
+        $data = $data->merge($this->loadCounter());
+        $data->put('list_desa', $list_desa);
+        return view('pesan.compose_pesan', $data->all());
+    }
+
+    /**
+     * @throws \Illuminate\Validation\ValidationException
+     */
+    public function storeComposePesan(Request $request)
+    {
+        try {
+            $this->validate($request, [
+                'judul'    => 'required',
+                'das_data_desa_id' => 'required|exists:das_data_desa,id',
+                'text' => 'required'
+            ]);
+
+            DB::transaction(function () use ($request) {
+                $id = Pesan::create([
+                    'das_data_desa_id' => $request->get('das_data_desa_id'),
+                    'judul' => $request->get('judul'),
+                    'jenis' => Pesan::PESAN_KELUAR,
+                    'sudah_dibaca' => 1,
+                ])->id;
+
+                PesanDetail::create([
+                    'pesan_id' => $id,
+                    'text' => Purify::clean($request->get('text')),
+                    'pengirim' => 'kecamatan',
+                    'nama_pengirim' => 'kecamatan - '. auth()->user()->name,
+                ]);
+            });
+
+            return redirect()->route('pesan.keluar')->with('success', 'Pesan berhasil dikirim!');
+        } catch (\Exception $e) {
+            return back()->withInput()->with('error', 'Pesan gagal dikirim!. Detail: ' . $e->getMessage());
+        }
+    }
+
+    public function setArsipPesan(Request $request)
+    {
+        $pesan = Pesan::findOrFail($request->get('id'));
+        $pesan->diarsipkan = Pesan::MASUK_ARSIP;
+        if ($pesan->save()) {
+            return redirect()->route('pesan.arsip')->with('success', 'Pesan berhasil diarsipkan!');
+        } else {
+            return back()->withInput()->with('error', 'Pesan gagal diarsipkan!');
+        }
+    }
+
+    public function setMultipleReadPesanStatus(Request $request)
+    {
+        $array = json_decode($request->get('array_id'));
+        $pesan = Pesan::whereIn('id', $array)->update([
+            'sudah_dibaca' => Pesan::SUDAH_DIBACA
+        ]);
+
+        if ($pesan > 0) {
+            return back()->with('success', 'Pesan berhasil ditandai!');
+        } else {
+            return back()->withInput()->with('error', 'Pesan gagal ditandai!');
+        }
+    }
+
+    public function setMultipleArsipPesanStatus(Request $request)
+    {
+        $array = json_decode($request->get('array_id'));
+        $pesan = Pesan::whereIn('id', $array)->update([
+            'diarsipkan' => Pesan::MASUK_ARSIP
+        ]);
+
+        if ($pesan > 0) {
+            return back()->with('success', 'Pesan berhasil ditandai!');
+        } else {
+            return back()->withInput()->with('error', 'Pesan gagal diarsipkan!');
+        }
```

Comment:
```suggestion
        if ($pesan > 0) {
            return back()->with('success', 'Pesan berhasil ditandai!');
        }
        return back()->withInput()->with('error', 'Pesan gagal diarsipkan!');
```

Mas @apidong, bisa dibuat bgni